### PR TITLE
IAU authority and EPSG-agnostic fixes

### PR DESCRIPTION
**Pull request by aaime**
Set of changes to make the code work beyond the EPSG authority, and a factory for planetary CRSs.
Work in progress.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Review by ianturton — APPROVED** [review]
lgtm, no issue with it going in to supported land

File: modules/library/referencing/src/test/resources/org/geotools/referencing/test-data/scripts/Equirectangular.txt [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Test points for the Robinson projection
+//
+// (C) 2020, Open Source Geospatial Foundation (OSGeo)
```
Comment (dromagnoli) [inline modules/library/referencing/src/test/resources/org/geotools/referencing/test-data/scripts/Equirectangular.txt:3]: Should that be 2023?

**Review by dromagnoli — APPROVED** [review]

File: modules/plugin/iau-wkt/src/main/java/org/geotools/referencing/iau/IAUAuthorityFactory.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2005-2008, Open Source Geospatial Foundation (OSGeo)
```
[the remark this answers is not in the document]
Comment (aaime) [reply modules/plugin/iau-wkt/src/main/java/org/geotools/referencing/iau/IAUAuthorityFactory.java:5]: Probably not :rofl:

File: modules/library/referencing/src/test/resources/org/geotools/referencing/test-data/scripts/Equirectangular.txt (hunk 1 above)
Comment (aaime) [reply modules/library/referencing/src/test/resources/org/geotools/referencing/test-data/scripts/Equirectangular.txt:3]: Yep

**Pull request merged**